File: Python Project/RollingDice.py
#importing libraries
import random

#function to roll dice
def roll_dice():
    print("Welcome to the Dice Rolling Simulator!")
    #while loop to run rolling dice features
    while True:
        #asking user to press enter button to roll dice
        input("Press Enter to roll the dice...")
        #using randint to randomly choosing number beteen 1, 6
        dice_roll = random.randint(1, 6)
        print(f'You rolled a {dice_roll}')
        #while to ask user to play again or close application
        while True:
            #taking input from user and converting them into lower case
            play_again = input("Would you like to roll again? (yes/no): ").lower()
            if play_again == "yes" or play_again == "no":
                break
            else:
                print("Please enter a valid option (yes/no)")
        if play_again == "no":
            print("Thanks for playing!")
            break
#calling roll_dice function
roll_dice()

File: JavaScript Project/RollingDice.js
//function to roll the dice
function rollDice() {
    console.log("Welcome to the Dice Rolling Simulator!");
    //while loop to run the dice rolling features
    while (true) {
        //asking the user to press Enter to roll the dice
        prompt("Press Enter to roll the dice...");
        // Using Math.random() to randomly choose a number between 1 and 6
        const diceRoll = Math.floor(Math.random() * 6) + 1;
        console.log(`You rolled a ${diceRoll}`);

        let playAgain;
        //while loop to ask the user to play again or close the application
        while (true) {
            //taking input from the user and converting it to lower case
            playAgain = prompt("Would you like to roll again? (yes/no)");

            if (playAgain.toLowerCase() === "yes" || playAgain.toLowerCase() === "no") {
                break;
            } else {
                console.log("Please enter a valid option (yes/no)");
            }
        }
        if (playAgain.toLowerCase() === "no") {
            console.log("Thanks for playing!");
            break;
        }
    }
}
//calling the rollDice function
rollDice();
